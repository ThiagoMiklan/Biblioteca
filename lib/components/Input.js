"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var _Control = _interopRequireDefault(require("./Control"));

var props_obj = {
  definition: "string",
  type: "string",
  placeholder: "string",
  value: "string",
  readonly: "boolean",
  disabled: "boolean",
  onChangeEvent: "function"
};

var Input = function Input(props) {
  (0, _type_validations.validate)(props, props_obj, "Input");
  return assemble(props);
};

function assemble(props) {
  var code = "";

  if (props.definition != undefined) {
    if (props.definition.includes("is-loading")) {
      code = assembleInputLoading(props);
    } else {
      code = assembleInput(props);
    }
  } else {
    code = assembleInput(props);
  }

  return code;
}

function assembleInputLoading(props) {
  var definition_input = props.definition == undefined ? "" : props.definition;
  var definition = 'control ' + definition_input;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: definition
  }, assembleInput(props));
}

function assembleInput(props) {
  var input_code = "";
  var definition = props.definition == undefined ? 'input' : "input " + props.definition;

  if (props.disabled) {
    if (props.readonly) {
      input_code = /*#__PURE__*/_react["default"].createElement(_Control["default"], null, /*#__PURE__*/_react["default"].createElement("input", (0, _extends2["default"])({
        onChange: props.onChangeEvent
      }, props.custom, {
        className: definition,
        type: props.type,
        placeholder: props.placeholder,
        value: props.value,
        disabled: true,
        readonly: true
      })));
    } else {
      input_code = /*#__PURE__*/_react["default"].createElement(_Control["default"], null, /*#__PURE__*/_react["default"].createElement("input", (0, _extends2["default"])({
        onChange: props.onChangeEvent
      }, props.custom, {
        className: definition,
        type: props.type,
        placeholder: props.placeholder,
        value: props.value,
        disabled: true
      })));
    }
  } else {
    input_code = /*#__PURE__*/_react["default"].createElement(_Control["default"], null, /*#__PURE__*/_react["default"].createElement("input", (0, _extends2["default"])({
      onChange: props.onChangeEvent
    }, props.custom, {
      className: definition,
      type: props.type,
      placeholder: props.placeholder,
      value: props.value
    })));
  }

  return input_code;
}

var _default = Input;
exports["default"] = _default;