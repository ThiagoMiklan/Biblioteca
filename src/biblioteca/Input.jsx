// @flow
import React, { useState } from 'react'
import PropTypes from 'prop-types';

type Props = {
    definition?: string,
    type?: string,
    placeholder?: string,
    value?: string,
    readonly?: bool,
    disabled?: bool,
    onChangeEvent?: () => void
}

const Input = (props: Props) => {
    return (
        assemble(props)
    );
}

function assemble(props) {
    var code = "";

    if (props.definition != undefined) {
        if (props.definition.includes("is-loading")) {
            code = assembleInputLoading(props);
        }else{
            code = assembleInput(props);
        }
    } else {
        code = assembleInput(props);
    }

    return code;

}
function assembleInputLoading(props) {
    var definition_input = (props.definition == undefined) ? "" : props.definition;
    var definition = 'control '+ definition_input;
    return <div className={definition}>
        {assembleInput(props)}

    </div>
}

function assembleInput(props) {
    var input_code = "";
    var definition = (props.definition == undefined )?'input': "input " + props.definition;

    if (props.disabled) {
        if (props.readonly) {
            input_code =
                <div className="control">
                    <input onChange={props.onChangeEvent} className={definition} type={props.type} placeholder={props.placeholder} value={props.value} disabled readonly></input>

                </div>
        } else {
            input_code =
                <div className="control">
                    <input onChange={props.onChangeEvent} className={definition} type={props.type} placeholder={props.placeholder} value={props.value} disabled></input>

                </div>
        }
    } else {
        input_code =
            <div className="control">
                <input onChange={props.onChangeEvent} className={definition} type={props.type} placeholder={props.placeholder} value={props.value}></input>

            </div>
    }
    return input_code;
}

Input.propTypes = {
    definition: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    value: PropTypes.string,
    readonly: PropTypes.bool,
    onChangeEvent: PropTypes.func
};


export default Input;