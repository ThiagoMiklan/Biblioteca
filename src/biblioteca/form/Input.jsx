import React,{useState} from 'react'
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassName from '../../tools/getClassName';
import { renderToString } from 'react-dom/server';

// sem icon ainda
const Input = (props) => {
    
    return (
        assemble(props)
    );
}

function createHook(){

}

function assemble(props) {
    var code = "";

    if (props.loading) {
        code = assembleInputLoading(props);
    } else {
        code = assembleInput(props);

    }
    return code;

}
function assembleInputLoading(props) {
    var definition = 'control is-loading ' + classnames(getClassName(props.definition, "Input"));
    return <div className={definition}>
        {assembleInput(props)}
        {props.icon}
    </div>
}

function assembleInput(props) {
    var input_code = "";
    var definition = 'input ' + classnames(getClassName(props.definition, "Input"));

    if (props.disabled) {
        if (props.readonly) {
            input_code =
                <div className="control">
                    <input onChange={props.onChangeEvent}  className={definition} type={props.type} placeholder={props.placeholder} value={props.value} disabled readonly></input>
                    {props.icon}
        </div>
        } else {
            input_code =
                <div className="control">
                    <input  onChange={props.onChangeEvent}  className={definition} type={props.type} placeholder={props.placeholder} value={props.value} disabled></input>
                    {props.icon}
        </div>
        }
    } else {
        input_code =
            <div className="control">
                <input onChange={props.onChangeEvent}  className={definition} type={props.type} placeholder={props.placeholder} value={props.value}></input>
                {props.icon}
        </div>
    }
    return input_code;
}

Input.propTypes = {
    definition: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    value: PropTypes.string,
    loading: PropTypes.bool,
    readonly: PropTypes.bool,
    onChangeEvent : PropTypes.func

};


export default Input;