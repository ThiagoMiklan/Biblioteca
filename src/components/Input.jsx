// @flow
import React, { useState } from 'react'
import { validate } from '../tools/type_validations.js';
import Control from './Control';

type Props = {
    definition?: string,
    type?: string,
    placeholder?: string,
    value?: string,
    readonly?: bool,
    disabled?: bool,
    onChangeEvent?: () => void
}

const props_obj = {
    definition: "string",
    type: "string",
    placeholder: "string",
    value: "string",
    readonly: "boolean",
    disabled: "boolean",
    onChangeEvent: "function"
}


const Input = (props: Props) => {
    validate(props, props_obj,"Input");
    return (
        assemble(props)
    );
}

function assemble(props) {
    var code = "";

    if (props.definition != undefined) {
        if (props.definition.includes("is-loading")) {
            code = assembleInputLoading(props);
        } else {
            code = assembleInput(props);
        }
    } else {
        code = assembleInput(props);
    }

    return code;

}
function assembleInputLoading(props) {
    var definition_input = (props.definition == undefined) ? "" : props.definition;
    var definition = 'control ' + definition_input;
    return <div className={definition}>
        {assembleInput(props)}

    </div>
}

function assembleInput(props) {
    var input_code = "";
    var definition = (props.definition == undefined) ? 'input' : "input " + props.definition;

    if (props.disabled) {
        if (props.readonly) {
            input_code =
                <Control>
                    <input onChange={props.onChangeEvent} className={definition} type={props.type} placeholder={props.placeholder} value={props.value} disabled readonly></input>
                </Control>
        } else {
            input_code =
                <Control>
                    <input onChange={props.onChangeEvent} className={definition} type={props.type} placeholder={props.placeholder} value={props.value} disabled></input>
                </Control>
        }
    } else {
        input_code =
            <Control>
                <input onChange={props.onChangeEvent} className={definition} type={props.type} placeholder={props.placeholder} value={props.value}></input>
            </Control>
    }
    return input_code;
}

export default Input;