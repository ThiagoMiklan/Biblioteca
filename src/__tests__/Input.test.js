import React from 'react';
import Input from '../components/Input'
import renderer from 'react-test-renderer';

test('input colors', () => {
    const input = renderer.create(
        <>
          <Input definition="is-primary" type="password" placeholder="Primary input"/>
    <Input definition="is-info" type="text" placeholder="Info input"/>
    <Input definition="is-success" type="text" placeholder="Success input"/>
    <Input definition="is-warning" type="text" placeholder="Warning input"/>
    <Input definition="is-danger" type="text" placeholder="Danger input"/>
        </>
    ).toJSON();

    expect(input).toMatchSnapshot();
})

test('input sizes', () => {
    const input = renderer.create(
        <>
           <Input definition="is-small" type="text" placeholder="Small input"/>
    <Input definition="is-normal" type="text" placeholder="Normal input"/>
    <Input definition="is-medium" type="text" placeholder="Medium input"/>
    <Input definition="is-large" type="text" placeholder="Large input"/>
        </>
    ).toJSON();

    expect(input).toMatchSnapshot();
})

test('input rounded', () => {
    const input = renderer.create(
        <>
             <Input definition="is-rounded" type="text" placeholder="Rounded input"/>
        </>
    ).toJSON();

    expect(input).toMatchSnapshot();
})

test('input loading sizes', () => {
    const input = renderer.create(
        <>
            <Input definition="is-loading is-small" type="text" placeholder="Loading input small" />
    <Input definition="is-loading" type="text" placeholder="Loading input normal"/>
    <Input definition="is-loading is-medium" type="text" placeholder="Loading input medium" />
    <Input definition="is-loading is-large" type="text" placeholder="Loading input large"/>
        </>
    ).toJSON();

    expect(input).toMatchSnapshot();
})

test('input disabled', () => {
    const input = renderer.create(
        <>
              <Input definition="is-loading is-large" type="text" placeholder="Loading input large" disabled={true}/>
        </>
    ).toJSON();

    expect(input).toMatchSnapshot();
})

