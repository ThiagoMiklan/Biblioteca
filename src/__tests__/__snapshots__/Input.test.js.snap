// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input colors 1`] = `
Array [
  <div
    className="control"
  >
    <input
      className="input is-primary"
      placeholder="Primary input"
      type="password"
    />
  </div>,
  <div
    className="control"
  >
    <input
      className="input is-info"
      placeholder="Info input"
      type="text"
    />
  </div>,
  <div
    className="control"
  >
    <input
      className="input is-success"
      placeholder="Success input"
      type="text"
    />
  </div>,
  <div
    className="control"
  >
    <input
      className="input is-warning"
      placeholder="Warning input"
      type="text"
    />
  </div>,
  <div
    className="control"
  >
    <input
      className="input is-danger"
      placeholder="Danger input"
      type="text"
    />
  </div>,
]
`;

exports[`input disabled 1`] = `
<div
  className="control is-loading is-large"
>
  <div
    className="control"
  >
    <input
      className="input is-loading is-large"
      disabled={true}
      placeholder="Loading input large"
      type="text"
    />
  </div>
</div>
`;

exports[`input loading sizes 1`] = `
Array [
  <div
    className="control is-loading is-small"
  >
    <div
      className="control"
    >
      <input
        className="input is-loading is-small"
        placeholder="Loading input small"
        type="text"
      />
    </div>
  </div>,
  <div
    className="control is-loading"
  >
    <div
      className="control"
    >
      <input
        className="input is-loading"
        placeholder="Loading input normal"
        type="text"
      />
    </div>
  </div>,
  <div
    className="control is-loading is-medium"
  >
    <div
      className="control"
    >
      <input
        className="input is-loading is-medium"
        placeholder="Loading input medium"
        type="text"
      />
    </div>
  </div>,
  <div
    className="control is-loading is-large"
  >
    <div
      className="control"
    >
      <input
        className="input is-loading is-large"
        placeholder="Loading input large"
        type="text"
      />
    </div>
  </div>,
]
`;

exports[`input rounded 1`] = `
<div
  className="control"
>
  <input
    className="input is-rounded"
    placeholder="Rounded input"
    type="text"
  />
</div>
`;

exports[`input sizes 1`] = `
Array [
  <div
    className="control"
  >
    <input
      className="input is-small"
      placeholder="Small input"
      type="text"
    />
  </div>,
  <div
    className="control"
  >
    <input
      className="input is-normal"
      placeholder="Normal input"
      type="text"
    />
  </div>,
  <div
    className="control"
  >
    <input
      className="input is-medium"
      placeholder="Medium input"
      type="text"
    />
  </div>,
  <div
    className="control"
  >
    <input
      className="input is-large"
      placeholder="Large input"
      type="text"
    />
  </div>,
]
`;
